### PR TITLE
Implement tracking on Fraud & Risk tools

### PR DESCRIPTION
**Pull request by allie500**
Fixes [#2867 on the server](https://github.com/Automattic/woocommerce-payments-server/issues/2867)

#### Changes proposed in this Pull Request

This PR adds the following tracking events to the WCPay fraud and risk tools: 
- wcpay_fraud_protection_banner_rendered
- wcpay_fraud_protection_banner_learn_more_button_clicked
- wcpay_fraud_protection_banner_remind_later_button_clicked
- wcpay_fraud_protection_tour_clicked_through
- wcpay_fraud_protection_tour_abandoned
- wcpay_fraud_protection_risk_level_preset_enabled
- wcpay_fraud_protection_advanced_settings_saved
- wcpay_fraud_protection_transaction_reviewed_merchant_blocked
- wcpay_fraud_protection_transaction_reviewed_merchant_approved
- wcpay_fraud_protection_basic_modal_viewed
- wcpay_fraud_protection_standard_modal_viewed
- wcpay_fraud_protection_high_modal_viewed
- wcpay_fraud_protection_advanced_settings_card_avs_mismatch_viewed
- wcpay_fraud_protection_advanced_settings_card_cvc_verification_viewed
- wcpay_fraud_protection_advanced_settings_card_international_ip_address_card_viewed
- wcpay_fraud_protection_advanced_settings_card_ip_address_mismatch_card_viewed
- wcpay_fraud_protection_advanced_settings_card_address_mismatch_viewed
- wcpay_fraud_protection_advanced_settings_card_price_threshold_viewed
- wcpay_fraud_protection_advanced_settings_card_items_threshold_viewed
- wcpay_fraud_protection_order_details_link_clicked

#### Testing instructions

* Check out this branch.
* All tests should pass.
* Make sure the fraud and risk tools feature flag option is set to `1`: `npm run wp option update wcpay_fraud_protection_settings_active 1`.
* Delete the banner and tour related options:
  * `npm run wp option delete wcpay_frt_discover_banner_settings`
  * `npm run wp option delete wcpay_fraud_protection_welcome_tour_dismissed`

**Flow to trigger events:** 
1. Visit the WCPay overview page. You should see the discoverability banner. 
2. Click the Learn more button. 
3. Go through all steps of the tour and click the Got it button on the final tour step.
4. Delete the tour option: `npm run option delete wcpay_fraud_protection_welcome_tour_dismissed`
5. Refresh the WCPay settings page, the tour should load again. This time click the 'X' button to close the tour.
6. Click the question mark icon next to the Standard protection level to view the Standard rules modal.
7. Click the question mark icon next to the High protection level to view the High rules modal.
8. Select each fraud protection preset radio button (basic, standard, and high) and click save settings button after you select each.
9. Select advanced protection radio button and click configure. 
10. On the configure page view all the option cards and make changes to the values (toggle on or off, set threshold values, etc). Then save settings.
11. Go back to the WCPay settings and set the protection level to high and save settings (just for ease of testing transactions in upcoming steps).
12. Go back to the WCPay overview page and click the 'Remind me later' button on the discoverability banner. 
13. With your protection level set to high, open a private browser window and perform the following transactions:
14. Perform _two_ separate test transactions with only one item in the order. These orders should be held for review.
15. Perform _two_ separate test transactions with over $1,000 USD total order value. These orders should be blocked and you should see an error notice on the checkout page for each attempted transaction.
16. Go to the orders page in WC admin: `/wp-admin/edit.php?post_type=shop_order`.
17. Click the first On-Hold order to view it. On the order details page, click the 'Review payment' link in the Fraud and Risk meta box:

Screenshot on 2023-03-31 at 09_22_24

18. On that transaction details page, click the 'Approve transaction' button.
19. Go back to the orders page and click the second On-Hold order.
20. This time, on the order details page click the 'View more details' link in the order note regarding the order being held for review.

Screenshot on 2023-03-31 at 09_25_18

21. This time, on this order's transaction details page click the 'Block transaction' button.
22. Now go back to the orders page and open the two blocked orders, they should show status: Pending payment.
23. On one of the blocked orders click the 'View more details' in the _Fraud & Risk meta box_. On the other order click the 'View more details' link in the order note regarding the order being blocked.

Screenshot on 2023-03-31 at 09_28_54

Screenshot on 2023-03-31 at 09_29_34

**Check Tracks Live View**

_Note: In my testing I have found that some events can take longer than 15 minutes to show up._

Search for the following event pattern: `wcadmin_wcpay_fraud_protection%`. You should see at least one of each event in the search results. For some events you will see multiple events since we triggered them multiple times.

Click the right arrow in the Props column on the left to view the event properties. The following events should have passed custom event properties:

- wcpay_fraud_protection_risk_level_preset_enabled
- wcpay_fraud_protection_advanced_settings_saved
- wcpay_fraud_protection_order_details_link_clicked

The first event should include what preset level was saved. The second event should include the advanced settings selected and saved. The final event should include the transactions status (review or blocked) and the type of link clicked (meta box or order note).

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5820` or branch name `add/2867-add-fraud-risk-tools-tracking` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=add/2867-add-fraud-risk-tools-tracking) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 9434a6c4a15dbeec62c347500dfdbcbc23b709d7
- Build time: 2023-03-31 18:41:13 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/payment-details/index.tsx [hunk 1]
```diff
@@ -18,6 +20,20 @@ const PaymentDetails: React.FC< PaymentDetailsProps > = ( { query } ) => {
...
 
+	const { status_is: statusIs, type_is: typeIs } = getQuery();
+
+	if ( statusIs && typeIs ) {
+		wcpayTracks.recordEvent(
+			'wcpay_fraud_protection_order_details_link_clicked',
+			{ status: statusIs, type: typeIs }
+		);
+
+		updateQueryString( {
```
Comment (tpaksu) [inline client/payment-details/index.tsx:31]: Can we add a comment here in the newly added block, so people don't remove the block mistakenly?

File: client/settings/fraud-protection/advanced-settings/index.js [hunk 2]
```diff
@@ -36,6 +36,38 @@ import './../style.scss';
 
 import { ProtectionLevel } from './constants';
 import { readRuleset, writeRuleset } from './utils';
+import wcpayTracks from 'tracks';
+
+const observerMapping = {
+	'avs-mismatch-card': {
```
Comment (tpaksu) [inline client/settings/fraud-protection/advanced-settings/index.js]: If all is going to have one single property, why don't we just rename it to `observerEventMapping` and get rid of the inner objects, just leave it with a key-value object?

File: client/settings/fraud-protection/advanced-settings/index.js [hunk 3]
```diff
@@ -146,6 +188,45 @@ const FraudProtectionAdvancedSettingsPage = () => {
...
+				// element is at least partially visible.
+				const { id } = target;
+				const { event } = observerMapping[ id ] || {};
+
+				if ( event ) {
+					wcpayTracks.recordEvent( event );
+				}
+
+				cardObserver.current?.unobserve(
+					document.getElementById( target.id )
```
Comment (tpaksu) [inline client/settings/fraud-protection/advanced-settings/index.js]: You already have `id` destructured from `target` above, we can just use `id` here.

Comment (allie500) [reply client/settings/fraud-protection/advanced-settings/index.js:206]: Yes. I was supposed to change that after Eduardo and I paired up to work on this. Thanks for catching it.

**Review by tpaksu — APPROVED** [review]
I saw all the tracks listed in the tracks live view, so approving in advance! The findings are all nitpicks, feel free to address them. LGTM!

File: client/settings/fraud-protection/advanced-settings/index.js [hunk 4]
```diff
@@ -196,14 +182,14 @@ const FraudProtectionAdvancedSettingsPage = () => {
 			if ( 0 < intersectionRatio ) {
 				// element is at least partially visible.
 				const { id } = target;
-				const { event } = observerMapping[ id ] || {};
+				const event = observerEventMapping[ id ] || {};
```
Comment (tpaksu) [inline client/settings/fraud-protection/advanced-settings/index.js]: You forgot to replace the `{}` to `null` or `false`.

Comment (allie500) [reply client/settings/fraud-protection/advanced-settings/index.js]: Good catch. Thank you. Fixed in [9434a6c](https://github.com/Automattic/woocommerce-payments/pull/5820/commits/9434a6c4a15dbeec62c347500dfdbcbc23b709d7).

File: client/settings/fraud-protection/advanced-settings/index.js (hunk 3 above)
Comment (allie500) [reply client/settings/fraud-protection/advanced-settings/index.js]: Thanks for catching this. I had mentioned it when Eduardo and I paired up on this, but I forgot to circle back to update it. Fixed in [498643a](https://github.com/Automattic/woocommerce-payments/pull/5820/commits/498643a79759ac936ecc33e22f2e96ae6eb7a2e0).

File: client/settings/fraud-protection/advanced-settings/index.js (hunk 2 above)
Comment (allie500) [reply client/settings/fraud-protection/advanced-settings/index.js]: Thanks for the suggestion. This is cleaner. Fixed in [498643a](https://github.com/Automattic/woocommerce-payments/pull/5820/commits/498643a79759ac936ecc33e22f2e96ae6eb7a2e0).

File: client/payment-details/index.tsx (hunk 1 above)
Comment (allie500) [reply client/payment-details/index.tsx:31]: Ok, I added this about `updateQueryString`:
> // Remove the tracking queries on page load so we don't track refreshes or back button reloads.

Fixed in [498643a](https://github.com/Automattic/woocommerce-payments/pull/5820/commits/498643a79759ac936ecc33e22f2e96ae6eb7a2e0).

**Pull request merged**